### PR TITLE
Refactor validate_query()

### LINKED ISSUE #156: QueryValidator in `das-serverless-function` does not accept a list of queries, therefore not allowing the use of a conjunction query (AND clause).

### PR DESCRIPTION
**Pull request by marcocapozzoli**
Resolves #156

### CODE REVIEW

**Review by andre-senna — APPROVED** [review]
Please wait for @levisingularity and @eddiebrissow approvals before merging

**Review by rafaellevidomuso — APPROVED** [review]

**Review by levisingularity — APPROVED** [review]

File: das-query-engine/validators/actions.py [hunk 1]
```diff
@@ -53,28 +53,31 @@ class QueryValidator(PayloadValidator):
...
-        if atom_type == "node" and not isinstance(name, str):
-            return False
-
-        if atom_type == "link" and not isinstance(targets, list):
-            return False
+            if atom_type == "link" and not isinstance(targets, list):
+                return False
 
-        return True
+            return True
```
Comment (Pedrobc89) [inline das-query-engine/validators/actions.py:80]: I believe this should be outside the for.
```suggestion
        return True
```

**Review by Pedrobc89 — CHANGES_REQUESTED** [review]

File: das-query-engine/tests/integration/handle/test_query_action.py [hunk 2]
```diff
@@ -33,12 +35,45 @@ def valid_event(self, action_type):
                 },
             }
         }
+    
+    @pytest.fixture
+    def query_list(self, action_type):
```
Comment (Pedrobc89) [inline das-query-engine/tests/integration/handle/test_query_action.py:40]: Success are good an all for testing, but testing the invalid paths are very important as well.
If so, you would've caught the `return True` inside the for.

File: das-query-engine/validators/actions.py [hunk 3]
```diff
@@ -53,28 +53,31 @@ class QueryValidator(PayloadValidator):
     strict = True
 
     @staticmethod
-    def validate_query(query, *args, **kwargs) -> bool:
-        if not isinstance(query, dict):
+    def validate_query(queries, *args, **kwargs) -> bool:
+        if isinstance(queries, dict):
+            queries = [queries]
+        elif not isinstance(queries, list) or not all(isinstance(query, dict) for query in queries):
```
Comment (eddiebrissow) [inline das-query-engine/validators/actions.py:59]: This limits the query, such as `[{}, []] `or `[[], []]`. To ensure the parameters are correct, we can to perform a recursive check and run additional tests, or simply verify the top-level type as either a list or a dictionary.

Comment (andre-senna) [reply das-query-engine/validators/actions.py:59]: > simply verify the top-level type as either a list or a dictionary.

This sounds like a good plan for now. @angeloprobst , do you have any inputs here?

Comment (angeloprobst) [reply das-query-engine/validators/actions.py:59]: I think that a recursive validation like the one suggested by @Pedrobc89 in the [ticket](https://github.com/singnet/das-serverless-functions/issues/156) covers all scenarios.

**Pull request merged**